src/troblueshooting/content.md

```diff
@@ -0,0 +1,551 @@
+- `8080` is the port exposed by the Service
+
+## Troubleshooting Ingress
+
+If you've reached this section:
+
+- the Pods are _Running_ and _Ready_
+- the Service distributes the traffic to the Pod
+
+But you still can't see a response from your cluster.
+
+At this point, the Ingress is likely misconfigured.
+
+Since the Ingress is a component that doesn't come by default in the cluster, there are different debugging techniques depending on the type of Ingress.
+
+But before diving into Ingress specific tools, there's something straightforward that you could check.
+
+The Ingress uses the `serviceName` and `servicePort` to connect to the Service.
+
+You should check that those are correctly configured.
+
+You can inspect that the Ingress is correctly configured with:
+
+```terminal|command=1|title=bash
+kubectl describe ingress <ingres-name>
+```
+
+If the _Backend_ column is empty, then there must be an error in the configuration.
+
+If you can see the endpoints in the _Backend_ column, but still can't route the traffic, the issue is likely to be:
+
+- how you exposed your Ingress to the public internet
+- how you exposed your cluster to the public internet
+
+You can isolate infrastructure issues from Ingress by connecting to the Ingress Pod directly.
+
+First, retrieve the Pod for your Ingress controller (which could be located in a different namespace):
+
+```terminal|command=1|title=bash
+kubectl get pods --all-namespaces
+NAMESPACE   NAME                              READY STATUS
+kube-system coredns-5644d7b6d9-jn7cq          1/1   Running
+kube-system etcd-minikube                     1/1   Running
+kube-system kube-apiserver-minikube           1/1   Running
+kube-system kube-controller-manager-minikube  1/1   Running
+kube-system kube-proxy-zvf2h                  1/1   Running
+kube-system kube-scheduler-minikube           1/1   Running
+kube-system nginx-ingress-controller-6fc5bcc  1/1   Running
+```
+
+Describe it to retrieve the port:
+
+```terminal|command=1|title=bash
+kubectl describe pod nginx-ingress-controller-6fc5bcc
+ --namespace kube-system \
+ | grep Ports
+```
+
+Finally, connect to the Pod:
+
+```terminal|command=1|title=bash
+kubectl port-forward nginx-ingress-controller-6fc5bcc 3000:80 --namespace kube-system
+```
+
+At this point, every time you visit port 3000 on your computer, the request is forwarded to port 80 on the Pod.
+
+_Does it works now?_
+
+- Yes, it worked! The issue is in the infrastructure. You should investigate how the traffic is routed to your cluster.
+- No, it doesn't work. The issue is in the Ingress controller. You should debug the Ingress
+
+If you still can't get the Ingress controller to work, you should start debugging it.
+
+Please notice that the Ingress controller comes in a different version.
+
+Popular options include Nginx, HAProxy, Traefik, etc.
+
+You should consult the official documentation for the project to find a troubleshooting guide specific to your Ingress controller.
```

Comment:
You should consult the documentation of your Ingress controller to find a troubleshooting guide.